controllers/front/AuthController.php

```diff
@@ -41,49 +40,25 @@ public function checkAccess()
 
     public function initContent()
     {
-        $should_redirect = false;
-
-        if (Tools::isSubmit('submitCreate') || Tools::isSubmit('create_account')) {
-            $register_form = $this
-                ->makeCustomerForm()
-                ->setGuestAllowed(false)
-                ->fillWith(Tools::getAllValues());
-
-            if (Tools::isSubmit('submitCreate')) {
-                $hookResult = array_reduce(
-                    Hook::exec('actionSubmitAccountBefore', [], null, true),
-                    function ($carry, $item) {
-                        return $carry && $item;
-                    },
-                    true
-                );
-                if ($hookResult && $register_form->submit()) {
-                    $should_redirect = true;
-                }
-            }
+        if (Tools::isSubmit('create_account')) {
+            return $this->redirectWithNotifications('registration');
```

Comment:
I guess the only thing that is left to handle, as we discussed in demo, is to handle this redirection with POST correctly right?

Reply:
So here are a few solutions amongst the one we discussed:
- perform a redirection with appropriate HTTP code to avoid losing POST method, not sure it works based on https://stackoverflow.com/questions/2604530/a-good-way-to-redirect-with-a-post-request But you can test if 301 or 302 works, seems like 307 should be the best candidate but depending on browsers it can remove the POST parameters..
- if the HTTP code doesn't work perform an internal forward using the new controller (see suggestion below)

Reply:
```suggestion
            $controller = new RegistrationController();
            
            return $tontroller->initContent();
```

I'm not sure AT ALL if this implementation is correct or even functional 😅. but it gives the general idea

Reply:
@jolelievre This redirection works perfectly. We were talking about a situation when you do:
- `$this->context->link->getPageLink('authentication', true, null, ['create_account' => '1']);`
- Pass this URL to smarty for your login popup.
- Make a POST request to it.

We could fix it by returning a correct URL in getPageLink, WDYT?

Then the only way it could fail is, if somebody was POSTing to hardcoded `/login?create_account` for whatever reason. 😄

Reply:
Actually, the case with hardcoded URL is exactly what I had in mind In this case we need to handle the POST redirection So we are sure that this modification from your PR is fully backward compatible

And this way no need to handle a tricky edge case in `getPageLink`